packages/frontend/src/components/wallet/TokenAmount.js

```diff
@@ -28,21 +31,37 @@ const showFullAmount = (amount, decimals, symbol) =>
         ? `${formatTokenAmount(amount, decimals, decimals)} ${symbol}`
         : '';
 
-const TokenAmount = ({ token: { balance, onChainFTMetadata }, withSymbol = false, className, showFiatAmount = true, "data-test-id": testId }) => (
-    <div className={className} title={showFullAmount(balance, onChainFTMetadata?.decimals, onChainFTMetadata?.symbol)} data-test-id={testId}>
-        <div>
-            {balance
-                ? formatToken(balance, onChainFTMetadata?.decimals)
-                : <span className='dots' />
+const TokenAmount = ({ 
+    token: { balance, onChainFTMetadata, priceInUSD, isWhiteListed = true }, 
+    withSymbol = false, 
+    className, 
+    showFiatAmount = true, 
+    "data-test-id": testId 
+}) => {
+    const tokenBalance = balance && formatToken(balance, onChainFTMetadata?.decimals);
+    const tokenPrice = (tokenBalance && priceInUSD) && big(+tokenBalance * +priceInUSD).toFixed(2);
+    return (
+        <div className={className} style={{color: isWhiteListed ? '' : '#FF585D'}} title={showFullAmount(balance, onChainFTMetadata?.decimals, onChainFTMetadata?.symbol)} data-test-id={testId}>
+            <div>
+                {balance
+                    ? tokenBalance
+                    : <span className='dots' />
+                }
+                <span className='currency'>{withSymbol ? ` ${onChainFTMetadata?.symbol}` : null}</span>
+            </div>
+            {showFiatAmount &&
+                <div className='fiat-amount' style={{color: isWhiteListed ? '' : '#FF585D', display: 'flex', whiteSpace: 'normal'}}>
+                { tokenPrice 
+                    ? `≈ $${tokenPrice} USD`
+                    : <Translate id='tokenBox.priceUnavailable'/>
```

Comment:
The `TokenBox` component renders this component, and it already displays `tokenBox.priceUnavailable`; this would lead to us repeating the `priceUnavailable` text multiple times on our main fungible token component display list